gwlearn/base.py

```diff
@@ -228,6 +232,8 @@ def _is_binary(series: pd.Series) -> bool:
                 adjacency=_kernel_functions[self.kernel](weights._adjacency, bandwidth),
                 is_sorted=True,
             )
+        if self.include_focal:
+            weights = weights.assign_self_weight(1)
```

Comment:
@tayloroshan Am I right in assigning 1 as the self-weight on focal?

Reply:
I don't think there's an issue in doing so, but I do think it could occur organically depending on the kernel function

Reply:
That would require changes in libpysal.